### ISSUE TITLE
way for sml / obis to accept  1.8.0 instead of 1-0:1.8.0 (so no prefix)

### ISSUE THREAD

**Issue by evlo** [esphome/feature-requests#1867]
**Describe the problem you have/What new integration you would like**

I'm trying to migrate from tasmota, i just like esphome better, but the smart meter interface is still impossible to me as my meter uses obis codes as espohome recognises. Ie. "1-0:1.8.0" is what esphome except as obis, but "1.8.0" is what I get from electrical meter. Do you know how this could be solved with some workaround?

output from tuya
![image](https://user-images.githubusercontent.com/8813196/189961459-fb383ac9-2587-4dc1-ab76-1c5e34b2c658.png)

**Please describe your use case for this integration and alternatives you've tried:**

I tried various custom OBIS reader external components

**Additional context**

Also to wake up meter the string /?! (hex 2F3F210D0A) at 300 baud, but it seems that rest of the communication could be done on 9600

**Comment by nagyrobi** [comment]
Please ask on Discord for faster answers.

**Comment by TheJulianJES** [comment]
@evlo I'm basically having the same issue as you. Did you manage to solve/work around it?

FYI, also asked on Discord: https://discord.com/channels/429907082951524364/429907082955718657/1042909407564013638

**Comment by evlo** [comment]
https://github.com/evlo/esphome_ZPA-ZE312

*https://github.com/jplitza/esphome_components*

**Comment by TheJulianJES** [comment]
@evlo Yep, I actually found your component in the mean-time! My meter sends a lot of other data that cause "errors", but total consumption + current consumption is working fine now!
Thanks for your work!
